hive_generator/pubspec.yaml

```diff
@@ -8,15 +8,15 @@ environment:
   sdk: ">=2.12.0 <3.0.0"
 
 dependencies:
-  build: ^2.0.0
-  source_gen: ^1.0.0
+  build: ^2.1.0
+  source_gen: ^1.1.0
   hive: ^2.0.4
-  analyzer: ^1.0.0
-  source_helper: ^1.1.0
+  analyzer: ^2.1.0
+  source_helper: ^1.2.1
```

Comment:
1. I don't think we need to change versions for `build`, `source_gen`, `source_helper` because they have annotated with `^` which means they support next minor versions (`^1.0.0` means it supports from `1.0.0` to exclusively `2.0.0`).
2. Hive doesn't exclusively requires analyzer >=2.0.0, it should work with both 1.X.Y and 2.K.Z versions so I think it's best to set version to `>=1.0.0 <3.0.0`

Because changing lower version constrains is considered breaking change.

In overview here's my suggested version constraints: 

```dart
  build: ^2.0.0
  source_gen: ^1.0.0
  analyzer: ">=1.0.0 <3.0.0"
  source_helper: ^1.1.0
```

Reply:
You know the best